test/IntegrationTests/Helpers/HealthzHelper.cs

```diff
@@ -17,40 +17,42 @@
 using System;
 using System.Net;
 using System.Net.Http;
+using System.Threading;
 using System.Threading.Tasks;
 using Xunit.Abstractions;
 
 namespace IntegrationTests.Helpers;
 
 internal static class HealthzHelper
 {
-    public static async Task<bool> TestHealtzAsync(string healthzUrl, string logPrefix, ITestOutputHelper output)
+    public static async Task TestAsync(string healthzUrl, ITestOutputHelper output)
     {
-        output.WriteLine($"{logPrefix} healthz endpoint: {healthzUrl}");
+        output.WriteLine($"Testing healthz endpoint: {healthzUrl}");
         HttpClient client = new();
 
-        for (int retry = 0; retry < 5; retry++)
+        var intervalMilliseconds = 500;
+        var maxMillisecondsToWait = 15_000;
+        var maxRetries = maxMillisecondsToWait / intervalMilliseconds;
+        for (int retry = 0; retry < maxRetries; retry++)
         {
-            HttpResponseMessage response;
-
             try
             {
-                response = await client.GetAsync(healthzUrl);
-            }
-            catch (TaskCanceledException)
-            {
-                response = null;
-            }
+                var response = await client.GetAsync(healthzUrl);
+                if (response.StatusCode == HttpStatusCode.OK)
+                {
+                    return;
+                }
 
-            if (response?.StatusCode == HttpStatusCode.OK)
+                output.WriteLine($"Healthz endpoint retured HTTP status code: {response.StatusCode}");
+            }
+            catch (Exception ex)
             {
-                return true;
+                output.WriteLine($"Healthz endpoint call failed: {ex.Message}");
             }
 
-            output.WriteLine($"{logPrefix} healthz failed {retry + 1}/5");
-            await Task.Delay(TimeSpan.FromSeconds(4));
```

Comment:
I remember this could be problematic (discussed with @RassK  some time ago).

Reply:
`Thread.Sleep` does block the thread, so you are not competing with other async/awaits to run it. However, the possible problem for both, i.e.: failing to wake up at the deadline, is due to CPU oversubscription. This is not going to be fixed by `Thread.Sleep`. In general, async/wait is much preferable to blocking a thread. That said I don't think it will make a material difference here.